buildSrc/src/main/kotlin/com/datadog/gradle/plugin/jsonschema/PokoGenerator.kt

```diff
@@ -180,16 +184,16 @@ class PokoGenerator(
             docBuilder.add("\n")
         }
         enumBuilder.addProperty(
-            PropertySpec.builder(ENUM_CONSTRUCTOR_JSON_VALUE_NAME, String::class, KModifier.PRIVATE)
+            PropertySpec.builder(ENUM_CONSTRUCTOR_JSON_VALUE_NAME, jsonValueType, KModifier.PRIVATE)
                 .initializer(ENUM_CONSTRUCTOR_JSON_VALUE_NAME)
                 .build()
         )
-
+        val parameterFormat = resolveEnumConstantFormat(definition)
         definition.values.forEach { value ->
             enumBuilder.addEnumConstant(
                 value.enumConstantName(),
```

Comment:
Maybe we also need to change the `enumConstantName()` resolution in case the type is `Number`.  Having backticks surrounding all values makes this not user friendly

Instead of 
```kotlin
    enum class Island(
        private val jsonValue: Number
    ) {
        `4`(4),
        `8`(8),
        `15`(15),
        `16`(16),
        `23`(23),
        `42`(42);
    }
```

We could have

```kotlin
    enum class Island(
        private val jsonValue: Number
    ) {
        ISLAND_4(4),
        ISLAND_8(8),
        ISLAND_15(15),
        ISLAND_16(16),
        ISLAND_23(23),
        ISLAND_42(42);
    }
```